Added Data Parallelism tutorial in blitz

hunkim opened the pull request:
Data Parallelism tutorial is added.
- Changed the narrative style
- Added blitz (where is the optional step?)

Please take a look and give me lots of comments. @chsasank, CC: @soumith

chsasank commented:
Thanks, this looks great. 

> Added blitz (where is the optional step?)

Can you edit last few lines of cifar10_tutorial so that the readers know this is an optional next step?
It'll also be nice if title of this tutorial has 'Optional: ' prefix.

hunkim commented:
@chsasank Thanks! Changed the name to `Optional: Data Parallelism`

Added in the cifar10 tutorial: If you want to see even more MASSIVE speedup using all of your GPUs, please check out Optional: Data Parallelism.

chsasank commented on beginner_source/blitz/data_parallel_tutorial.py:
```diff
@@ -0,0 +1,254 @@
+# 
+# DataParallel splits your data automatically and sends job orders to multiple
+# models on several GPUs. After each model finishes their job, DataParallel
+# collects and merges the results before returning it to you. It's really awesome!
```
Can you remove it's awesome line? You know, it doesn't fit to the tone, I think.

chsasank commented on beginner_source/blitz/data_parallel_tutorial.py:
```diff
@@ -0,0 +1,254 @@
+"""
+Optional: Data Parallelism
+============
```
I see ` WARNING: Title underline too short.` during sphinx build. Can you fix all such warnings?

chsasank commented:
Thanks, merged!
